lib/web_ui/lib/ui_web/src/ui_web/navigation/platform_location.dart

```diff
@@ -0,0 +1,126 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:ui/src/engine.dart';
+
+import 'url_strategy.dart';
+
+/// Encapsulates all calls to DOM apis, which allows the [UrlStrategy] classes
+/// to be platform agnostic and testable.
+///
+/// For convenience, the [PlatformLocation] class can be used by implementations
+/// of [UrlStrategy] to interact with DOM apis like pushState, popState, etc.
+abstract class PlatformLocation {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const PlatformLocation();
```

Comment:
It looks like this class is meant to be implemented rather than extended, so we shouldn't need any constructor at all. Also, if we're allowed to use Dart 3 features, maybe we can make it `interface class` (or whatever the new syntax is for interfaces).